File: test/fakers/checkout/payment_faker.dart
import 'package:faker/faker.dart';
import 'package:sertton/core/checkout/dtos/payment_dto.dart';

class PaymentFaker {
  static final _faker = Faker();

  static PaymentDto fakeDto({
    String? id,
    String? name,
    String? icon,
    String? pdf,
    PaymentMethod? method,
  }) {
    return PaymentDto(
      id: id ?? _faker.guid.guid(),
      name: name ?? 'Credit Card',
      icon: icon ?? 'credit_card',
      pdf: pdf ?? _faker.internet.httpsUrl(),
      method: method ?? PaymentMethod.creditCard,
    );
  }
}

File: src/core/checkout/entities/fakers/payment-faker.ts
import { faker } from "@faker-js/faker"

import { type Payment, PaymentMethod } from "../payment"

export class PaymentFaker {
  static fake(payment?: Partial<Payment>): Payment {
    return {
      icon: faker.image.url(),
      id: faker.string.uuid(),
      method: faker.helpers.arrayElement(Object.values(PaymentMethod)),
      name: faker.finance.currencyName(),
      pdf: faker.internet.url(),
      ...payment,
    }
  }

  static fakeMany(count = 10, payment?: Partial<Payment>): Payment[] {
    return Array.from({ length: count }, () => PaymentFaker.fake(payment))
  }
}
